### PR TITLE
Update links to Lightbend RPv2, more warnings about auto-down

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-cluster/src/main/resources/reference.conf [hunk 1]
```diff
@@ -26,6 +26,10 @@ akka {
     # unreachable nodes as DOWN after a configured time of unreachability?
     # Using auto-down implies that two separate clusters will automatically be
     # formed in case of network partition.
+    #
+    # Don't enable this in production, see 'Auto-downing (DO NOT USE)' section 
+    # of Akka Cluster documentation.
```
Comment (ktoso) [inline akka-cluster/src/main/resources/reference.conf]: good

File: akka-cluster/src/main/scala/akka/cluster/AutoDown.scala [hunk 2]
```diff
@@ -62,6 +63,8 @@ private[cluster] class AutoDown(autoDownUnreachableAfter: FiniteDuration)
 
   // re-subscribe when restart
   override def preStart(): Unit = {
+    log.warning("Don't use auto-down feature of Akka Cluster in production. " +
+      "See 'Auto-downing (DO NOT USE)' section of Akka Cluster documentation.")
```
Comment (ktoso) [inline akka-cluster/src/main/scala/akka/cluster/AutoDown.scala]: good

**Comment by patriknw** [comment]
For the lurking reviewer RPv2 means that Lightbend's commercial features such as Split Brain Resolver and Monitoring can be used together with the ordinary OSS artifacts.

**Comment by patriknw** [comment]
@jducoeur This is the best I can do right now, unfortunately. Thanks for your help with spreading the word about the problem with auto-down.

**Comment by johanandren** [comment]
LGTM!

File: akka-docs/rst/java/cluster-sharding.rst [hunk 3]
```diff
@@ -25,6 +25,12 @@ to route the message with the entity id to the final destination.
 Cluster sharding will not be active on members with status :ref:`WeaklyUp <weakly_up_java>` 
 if that feature is enabled.
 
+.. warning::
+   **Don't use Cluster Sharding together with Automatic Downing**,
```
Comment (ktoso) [inline akka-docs/rst/java/cluster-sharding.rst]: This should explain though that we specifically mean auto-down,
now it sounds like any automatic one is doomed.

I would not mind a link to SBR here too

File: akka-docs/rst/java/cluster-singleton.rst [hunk 4]
```diff
@@ -73,7 +73,7 @@ Especially the last point is something you should be aware of — in general whe
 you should take care of downing nodes yourself and not rely on the timing based auto-down feature.
 
 .. warning::
-   **Be very careful when using Cluster Singleton together with Automatic Downing**,
+   **Don't use Cluster Singleton together with Automatic Downing**,
```
Comment (ktoso) [inline akka-docs/rst/java/cluster-singleton.rst]: 👍

File: akka-docs/rst/java/cluster-sharding.rst (hunk 3 above)
Comment (ktoso) [inline akka-docs/rst/java/cluster-sharding.rst]: Ok, nevermind - i see that docs section we link to here has a link now, that's good

**Pull request merged**